.travis.yml

```diff
@@ -52,15 +52,15 @@ jobs:
       - *no_optimize
       - *imports
 
-    - env: &py36_dev
+    - env: &py37_dev
```

Comment:
I'd prefer that this build was marked as an allowed failure - frequently upstream libraries make changes that break our test suite. While we should fix these (either by reporting fixes up stream, or by patching locally), they result in our tests being red for a long amount of time. A test suite that is always red, but has "known failures that we're ignoring for now" isn't useful. Marking it as an allowed failure allows for developers to try and fix these failures in the CI (tests are still run, can see results), but doesn't block normal development.

Reply:
@jcrist I agree with your point, a test suite that is always red isn't useful at all. However, having I feel that having an entire build that is allowed to fail or is manually triggered is probably just as useless as the previous case, it will certainly get forgotten/ignored by people.

Honestly, I don't know what's a well-balanced solution for this situation, which is very likely the reason why that development build got disabled in the first place.

I think the only possible uses for such a build is to either use it as a means to track down _when_ a change was introduced that broke something, or as a nightly build that may indeed fail without disturbing contributors with PRs failing all the time. Both cases would involve someone manually checking for issues frequently, which I don't know if it's realistically useful.

Reply:
I see a dev build as useful for having CI introspection when debugging fixes for upstream dependencies (say I want to use pyarrow dev with dask, it's nice to use the CI to double-check that the things I fix lead to things working with pyarrow dev). However, with the number of upstream libraries we care about, and the pace of development for these libraries, having a dev build has led to frequent breakages in the past. While we could try and stay on top of these, it's often easier just to wait for the next release and fix them all then. I suspect the latter approach uses less dev time overall (there is some overhead for each patch we make to keep up with e.g. pandas master).

In my comment [here](https://github.com/dask/dask/pull/4525#issuecomment-480867766) I suggested that we either make the build an allowed failure *or* use travis conditional build logic to make running it optional. This still allows for checking if changes fix existing upstream bugs without slowing down our development pace.

Reply:
Fair enough. I'd say having such a dev build running nightly (assuming Travis allows that, of course) would then be a useful thing to have, this way we can track how upstream is developing without us having to trigger that build manually.

Reply:
I suggest still allowing a way to run the build in a PR with a conditional build. We already do this for the hdfs tests: https://github.com/dask/dask/blob/c9285c589e1ca7dce11a5994805cd76e5b02c74b/.travis.yml#L71-L78

I think the following should work:

```
- env
  if: type == cron OR commit_message =~ test-upstream  # Run on cron jobs or any build where the commit message contains "test-upstream"
  ...
```

We'd also want to add:

```
- env:
  if: type != cron  # don't run on cron builds
  ...
```

to the other builds.

We can then add a cron build as described here: https://docs.travis-ci.com/user/cron-jobs/

Reply:
@jcrist thanks for the directions, and yes, this is what meant, to allow triggering during PR _and_ nightly.

@jakirkham @mrocklin what do you guys think of having this dev build running nightly in addition to the possibility of triggering it during PRs?

Reply:
Btw, I think we should also build the other builds nightly as well, I can't think of a reason not to at least.

Reply:
Per-commit status is always good, especially for auto-merged code. Just because a PR succeeded, doesn't mean that merged `master` will. :-)